File: solutions/python/difference-of-squares/2/difference_of_squares.py
def square_of_sum(count):
    return sum(range(1, count + 1)) ** 2


def sum_of_squares(count):
    return sum(n * n for n in range(1, count + 1))


def difference(count):
    return square_of_sum(count) - sum_of_squares(count)

File: solutions/dart/difference-of-squares/2/lib/difference_of_squares.dart
// dart 2.5.0
import 'dart:math' show pow;
Iterable<int> list(int n)=>Iterable<int>.generate(n, (i)=>i+1);
int sum(int result,int e) => result + e;
int square(int n) => n * n;

class DifferenceOfSquares {
  int squareOfSum(int firstNaturalNumber)=> square(list(firstNaturalNumber).reduce(sum));
  int sumOfSquare(int firstNaturalNumber)=> list(firstNaturalNumber).map(square).reduce(sum);
  int differenceOfSquares(int firstNaturalNumber) => squareOfSum(firstNaturalNumber) - sumOfSquare(firstNaturalNumber);
}
